zapcore/encoder_test.go

```diff
@@ -360,70 +393,93 @@ func TestEncoderConfiguration(t *testing.T) {
+				StacktraceKey:    "S",
+				LineEnding:       base.LineEnding,
+				EncodeTime:       base.EncodeTime,
+				EncodeDuration:   base.EncodeDuration,
+				EncodeLevel:      base.EncodeLevel,
+				EncodeCaller:     func(EntryCaller, PrimitiveArrayEncoder) {},
+				EncodeLoggerName: base.EncodeLoggerName,
 			},
 			expectedJSON:    `{"L":"info","T":0,"N":"main","C":"foo.go:42","M":"hello","S":"fake-stack"}` + "\n",
 			expectedConsole: "0\tinfo\tmain\thello\nfake-stack\n",
 		},
 		{
+			desc: "handle no-op EncodeLoggerName",
+			cfg: EncoderConfig{
+				LevelKey:         "L",
+				TimeKey:          "T",
+				MessageKey:       "M",
+				NameKey:          "N",
+				CallerKey:        "C",
+				StacktraceKey:    "S",
+				LineEnding:       base.LineEnding,
+				EncodeTime:       base.EncodeTime,
+				EncodeDuration:   base.EncodeDuration,
+				EncodeLevel:      base.EncodeLevel,
+				EncodeCaller:     base.EncodeCaller,
+				EncodeLoggerName: func(string, PrimitiveArrayEncoder) {},
+			},
+			expectedJSON:    `{"L":"info","T":0,"N":"main","C":"foo.go:42","M":"hello","S":"fake-stack"}` + "\n",
+			expectedConsole: "0\tinfo\tfoo.go:42\thello\nfake-stack\n",
+		},
+		{
 			desc: "use custom line separator",
 			cfg: EncoderConfig{
-				LevelKey:       "L",
-				TimeKey:        "T",
-				MessageKey:     "M",
-				NameKey:        "N",
-				CallerKey:      "C",
-				StacktraceKey:  "S",
-				LineEnding:     "\r\n",
-				EncodeTime:     base.EncodeTime,
-				EncodeDuration: base.EncodeDuration,
-				EncodeLevel:    base.EncodeLevel,
-				EncodeCaller:   base.EncodeCaller,
+				LevelKey:         "L",
+				TimeKey:          "T",
+				MessageKey:       "M",
+				NameKey:          "N",
+				CallerKey:        "C",
+				StacktraceKey:    "S",
+				LineEnding:       "\r\n",
+				EncodeTime:       base.EncodeTime,
+				EncodeDuration:   base.EncodeDuration,
+				EncodeLevel:      base.EncodeLevel,
+				EncodeCaller:     base.EncodeCaller,
+				EncodeLoggerName: base.EncodeLoggerName,
 			},
 			expectedJSON:    `{"L":"info","T":0,"N":"main","C":"foo.go:42","M":"hello","S":"fake-stack"}` + "\r\n",
 			expectedConsole: "0\tinfo\tmain\tfoo.go:42\thello\nfake-stack\r\n",
 		},
 		{
 			desc: "omit line separator definition - fall back to default",
 			cfg: EncoderConfig{
-				LevelKey:       "L",
-				TimeKey:        "T",
-				MessageKey:     "M",
-				NameKey:        "N",
-				CallerKey:      "C",
-				StacktraceKey:  "S",
-				EncodeTime:     base.EncodeTime,
-				EncodeDuration: base.EncodeDuration,
-				EncodeLevel:    base.EncodeLevel,
-				EncodeCaller:   base.EncodeCaller,
+				LevelKey:         "L",
+				TimeKey:          "T",
+				MessageKey:       "M",
+				NameKey:          "N",
+				CallerKey:        "C",
+				StacktraceKey:    "S",
+				EncodeTime:       base.EncodeTime,
+				EncodeDuration:   base.EncodeDuration,
+				EncodeLevel:      base.EncodeLevel,
+				EncodeCaller:     base.EncodeCaller,
+				EncodeLoggerName: base.EncodeLoggerName,
```

Comment:
We should only need to add two tests here: one that uses the capitalized encoder, and one that uses a no-op encoder. The other tests should pass as-is, and keeping them that way ensures that we don't break backward-compatibility later.